kite-data/kite-data-core/src/main/java/org/kitesdk/data/spi/AbstractRefinableView.java

```diff
@@ -223,4 +223,41 @@ public URI getUri() {
     }
     return builder.build();
   }
+
+  @Override
+  public void signalReady() {
+    Pair<DatasetRepository, Map<String, String>> pair =
+        Registration.lookupDatasetUri(URI.create(getUri().getRawSchemeSpecificPart()));
+    DatasetRepository repo = pair.first();
+
+    repo.signalReady(dataset.getNamespace(), dataset.getName(), getConstraints());
+  }
+
+  @Override
+  public boolean isReady() {
+    Pair<DatasetRepository, Map<String, String>> pair =
+        Registration.lookupDatasetUri(URI.create(getUri().getRawSchemeSpecificPart()));
+    DatasetRepository repo = pair.first();
```

Comment:
Currently it's fairly awkward to get back to the dataset repository that a dataset originated from. What do you guys think about adding a "getRepostiory" method to the Dataset API to it more natural to work back up the tree?